### PR TITLE
EKS automation fixes

### PR DESCRIPTION
**Pull request by aaronyeeski**
This PR addresses fixes for hosted EKS test cases.

- [x] Address errors seen in `test_eks_v2_hosted_cluster_create_all`
- [x] Address errors seen in `test_eks_v2_create_import_cluster`
- [x] Address errors seen in `test_eks_v2_hosted_cluster_delete`
- [ ] Handle EKS cluster provisioning for Rancher v2.5.6 and above.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 1]
```diff
@@ -165,16 +166,18 @@ def test_eks_v2_hosted_cluster_delete():
...
 
 
 @ekscredential
 def test_eks_v2_create_import_cluster():
     """
     Create an imported EKS cluster with some default values in EKS config
     """
-    display_name = create_resources_eks()
-    cluster_name = random_test_name("test-auto-eks")
+    # Temporarily commented, as botoclient is not working
```
Comment (aaronRancher) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:177]: Created a workaround for import test case until we can find a solution to botoclient errors.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 2]
```diff
@@ -190,6 +193,35 @@ def test_eks_v2_create_import_cluster():
...
+# EKS launch template test cases
+
+
+@ekscredential
+def test_eks_v2_hosted_cluster_create_basic_launch_template():
+    """
+    Create a hosted EKS v2 cluster with all default values from the UI
+    This does not use a preexisting launch template.
+    """
+    cluster_name = random_test_name("test-auto-eks")
```
Comment (aaronRancher) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:206]: This is a basic EKS cluster create using launch templates. We currently do not have a check in place to differentiate  cluster creation in v2.5.5 vs v2.5.6.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 3]
```diff
@@ -173,8 +174,10 @@ def test_eks_v2_create_import_cluster():
     """
     Create an imported EKS cluster with some default values in EKS config
     """
-    display_name = create_resources_eks()
-    cluster_name = random_test_name("test-auto-eks")
+    # Temporarily commented, as botoclient is not working
+    # display_name = create_resources_eks()
+    # cluster_name = random_test_name("test-auto-eks")
+    cluster_name = MANUAL_IMPORTED_EKS_CLUSTER
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:180]: Can we add an if condition - if MANUAL_IMPORTED_EKS_CLUSTER then use that, else create a cluster?

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 4]
```diff
@@ -266,7 +298,7 @@ def get_logging_types():
     return logging_types
 
 
-def get_eks_config_basic(cluster_name):
+def get_eks_config_basic(cluster_name, launch_template=False):
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:301]: Did we start adding code for Launch templates also? Maybe we can have that in the next PR ? We can have this PR to just fix the existing test cases

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py (hunk 2 above)
Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:206]: If so, can we add this in the next PR ?

**Review by sowmyav27 — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py (hunk 3 above)
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py]: Added check for MANUAL_IMPORTED_EKS_CLUSTER. If name is not found, cluster will be created using botoclient.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py (hunk 4 above)
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py]: I have moved launch template code out of this PR.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 5]
```diff
@@ -26,6 +26,8 @@
                                           'cannot create cluster')
 DEFAULT_TIMEOUT_EKS = 1200
 IMPORTED_EKS_CLUSTERS = []
+MANUAL_IMPORTED_EKS_CLUSTER = os.environ.get(
+                                'RANCHER_MANUAL_EKS_IMPORT', "")
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:30]: Can we name this as `RANCHER_EXISTING_EKS_CLUSTER`

**Review by sowmyav27 — APPROVED** [review]
LGTM

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 6]
```diff
@@ -207,11 +208,13 @@ def create_project_client(request):
...
         client = get_user_client()
-        for name, cluster in cluster_details.items():
-            if len(client.list_cluster(name=name).data) > 0:
-                client.delete(cluster)
-        for display_name in IMPORTED_EKS_CLUSTERS:
-            AmazonWebServices().delete_eks_cluster(cluster_name=display_name)
+        if CLEANUP_CLUSTER:
+            for name, cluster in cluster_details.items():
+                if len(client.list_cluster(name=name).data) > 0:
+                    client.delete(cluster)
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:214]: Do the resources get actually cleaned after the finalizer? Just to confirm.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 7]
```diff
@@ -174,13 +175,13 @@ def test_eks_v2_create_import_cluster():
     Create an imported EKS cluster with some default values in EKS config
     """
     display_name = create_resources_eks()
-    cluster_name = random_test_name("test-auto-eks")
-    eks_config_temp = get_eks_config_basic(cluster_name)
+    # cluster_name = random_test_name("test-auto-eks")
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:178]: Remove comment if this is not needed.

**Review by izaac — CHANGES_REQUESTED** [review]
A couple of comments. 
And I will also need to see the result/output on the freeform job (DM me the link)

**Pull request closed without merging**